### PR TITLE
Fix PHP7.4 bug webservice blank schemas for resources

### LINKED ISSUE #26559: [WS][1.7.8] Cannot receive blank schemas for resources
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

It's the same issue that #21748 and #19305

Here you have the PHP version 7.4 compatible with prestashop 1.7.8 but the webservice works with PHP 7.3 but not with 7.4 : 
https://devdocs.prestashop.com/1.7/basics/installation/system-requirements/#php-compatibility-chart

https://user-images.githubusercontent.com/6583978/164058319-cd3d59bd-bcc4-41d3-9676-c40ecae69a56.mov

### Expected behavior

Receive a blank body of resource

### Steps to reproduce

1) Install Prestashop 1.7.8.0 with PHP 7.4
2) Make GET call to an PrestaShop api : api/products?schema=blank
3) Disable the debug mode
4) See error

### PrestaShop version(s) where the bug happened

1.7.8.0

### PHP version(s) where the bug happened

7.4.2

### If your bug is related to a module, specify its name and its version

_No response_

### LINKED ISSUE #28279: PHP Notice Error while accessing API on /products
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

This error has bee previously reported for version 1.7.7 beta. The resolution was then that version 1.7.7 is not compatible with php 7.4. However, Prestashop 1.7.8 should be compatible with php 7.4.

### Expected behavior

_No response_

### Steps to reproduce

1. load page https://...webservice_address.../api/products?schema=blank
2. The result contains the error mentioned above, plus a similar error for line 719.
[products-response.txt](https://github.com/PrestaShop/PrestaShop/files/8506933/products-response.txt)
 

### PrestaShop version(s) where the bug happened

1.7.8.5

### PHP version(s) where the bug happened

7.4.3

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by preoteasa**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | The fix addresses a compatibility issue in webservice and php 7.4. This fixes issue #28279.
| Type?             | bug fix
| Category?         | WS
| BC breaks?        |  no
| Deprecations?     | no
| Fixed ticket?     | Fixes #26559
| Related PRs       | 
| How to test?      | Follow https://github.com/PrestaShop/PrestaShop/issues/26559 steps
| Possible impacts? | Webservice: creating a new product using the web service.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/28280)

### CODE REVIEW

**Review by matks — CHANGES_REQUESTED** [review]
Dear @preoteasa thanks for this pull request!

This PR aim is to fix a bug, so first we need to be able to reproduce the bug. A QA analyst tried to reproduce issue https://github.com/PrestaShop/PrestaShop/issues/28279 and failed. Can you help her ? We need to be able to reproduce the issue to validate the PR fixes it 😉

**Comment by preoteasa** [comment]
I have answered this on issue [#28279](https://github.com/PrestaShop/PrestaShop/issues/28279). It seems that, if debug mode is on, then the problem does not occur.

**Comment by hibatallahAouadni** [comment]
Thanks @preoteasa for your PR :rocket: 
I just edited your PR description to fix this issue #26559 as your issue has been closed as a duplicate :wink:

**Review by matks — DISMISSED** [review]
We have now a bug we can reproduce 👍 thank you I approve the PR.

**Review by atomiix — DISMISSED** [review]

File: classes/Product.php [hunk 1]
```diff
@@ -7069,7 +7069,7 @@ public function getCoverWs()
     {
         $result = $this->getCover($this->id);
 
-        return $result['id_image'];
+        return $result ? $result['id_image'] : null;
```
Comment (Progi1984) [inline classes/Product.php:7072]: Please update the PHPDoc :)

**Review by Progi1984 — COMMENTED** [review]
A small comment because I think PHPStan will be broken after the merge.

File: classes/Product.php (hunk 1 above)
Comment (preoteasa) [reply classes/Product.php:7072]: I don't understand what to add.

Comment (atomiix) [reply classes/Product.php:7072]: A few lines before, you have ` * @return int|string`, you should replace it with ` * @return int|string|null` or even ` * @return int|null` if you're sure `$result['id_image']` cannot be a `string`.

Comment (preoteasa) [reply classes/Product.php:7072]: I see now what you mean. I pushed the change. Null should be legal return value as it is accepted by functions calling getCoverWs

**Review by Progi1984 — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Comment by SD1982** [comment]
Hi @preoteasa your PR will be tested as soon as issue #28357 is fixed thanks for your patience

**Comment by Progi1984** [comment]
@SD1982 This PR is now unlocked.

**Review by AureRita — APPROVED** [review]
Hi @preoteasa 

Thank you for your PR, I tested it and as you can see, it works :

https://user-images.githubusercontent.com/100699445/168820121-cd435a3a-0cc6-4212-9cfe-529fe640d7f7.mp4

**Comment by kpodemski** [comment]
Thank you @preoteasa @AureRita

**Pull request merged**